acme/db/nosql/account.go

```diff
@@ -177,55 +178,106 @@ func (db *DB) CreateExternalAccountKey(ctx context.Context, name string) (*acme.
 	}
 
 	dbeak := &dbExternalAccountKey{
-		ID:        keyID,
-		Name:      name,
-		KeyBytes:  random,
-		CreatedAt: clock.Now(),
+		ID:              keyID,
+		ProvisionerName: provisionerName,
+		Name:            name,
+		KeyBytes:        random,
+		CreatedAt:       clock.Now(),
 	}
 
 	if err = db.save(ctx, keyID, dbeak, nil, "external_account_key", externalAccountKeyTable); err != nil {
 		return nil, err
 	}
 	return &acme.ExternalAccountKey{
-		ID:        dbeak.ID,
-		Name:      dbeak.Name,
-		AccountID: dbeak.AccountID,
-		KeyBytes:  dbeak.KeyBytes,
-		CreatedAt: dbeak.CreatedAt,
-		BoundAt:   dbeak.BoundAt,
+		ID:              dbeak.ID,
+		ProvisionerName: dbeak.ProvisionerName,
+		Name:            dbeak.Name,
+		AccountID:       dbeak.AccountID,
+		KeyBytes:        dbeak.KeyBytes,
+		CreatedAt:       dbeak.CreatedAt,
+		BoundAt:         dbeak.BoundAt,
 	}, nil
 }
 
 // GetExternalAccountKey retrieves an External Account Binding key by KeyID
-func (db *DB) GetExternalAccountKey(ctx context.Context, keyID string) (*acme.ExternalAccountKey, error) {
+func (db *DB) GetExternalAccountKey(ctx context.Context, provisionerName string, keyID string) (*acme.ExternalAccountKey, error) {
 	dbeak, err := db.getDBExternalAccountKey(ctx, keyID)
 	if err != nil {
 		return nil, err
 	}
 
+	if dbeak.ProvisionerName != provisionerName {
```

Comment:
Why is it important that the provisioner name in the request match the original provisionerName used to bind the key?

Reply:
For future reference and as discussed on Slack: a user can enable or disable EAB per provisioner. It made sense to me to also link creation of the credentials to a provisioner, so that there are less surprises.